full.async/src/full/async.clj

```diff
@@ -62,6 +68,26 @@
              (recur (dec retries#)))
            res#)))))
 
+(defmacro go-try> [err-chan & body]
```

Comment:
The general motivation is to have sane error management, as it can be very hairy in distributed systems. As Rich suggested, I had a look at error handling in erlang and as far as I understood I think it is well designed. The problem with go-loops et al. is that they are no named identities in your system like actors, so you cannot monitor them by address. I think the best way to decouple parts of the system then is by giving them a dedicated error channel to report on (with the help of the *-try macros) and then monitor for errors at subsystem boundaries and restart parts of the system as needed. That way one can start with a single dedicated error-channel, pass it in all systems to catch all errors (like a distributed try-catch) and then make the system robust by catching errors at appropriate boundaries and restart parts of the system.

For a single go routine it might not be as important (go-try>) as I usually block on it and its error anyway, I added it for consistency. But you cannot track errors in go-loops easily if they run concurrently.  These ideas are still wip, what is your take on error handling?